pyccel/ast/core.py

```diff
@@ -226,6 +233,101 @@ def handle_precedence(args, my_precedence):
 
     return args
 
+def bool2int(a):
+    if a.dtype is not NativeInteger():
+        return Int(a)
+    else:
+        return a 
+
+class PyccelBitOperator(Expr, PyccelAstNode):
+    _rank = 0
+    _shape = ()
+
+    def __init__(self, args):
+        if self.stage == 'syntactic':
+            self._args = handle_precedence(args, self.precedence)
+            return
+
+        max_precision = 0
+        for a in args:
+            if a.dtype is NativeInteger() or a.dtype is NativeBool():
+                max_precision = max(a.precision, max_precision)
+            else:
+                raise TypeError('unsupported operand type(s): {}'.format(self))
+        self._precision = max_precision
+    @property
+    def precedence(self):
+        return self._precedence
+
+class PyccelRShift(PyccelBitOperator):
+    _precedence = 11
+    _dtype = NativeInteger()
+    def __init__(self, *args):
+        super(PyccelRShift, self).__init__(args)
+        if self.stage == 'syntactic':
+            return
+        self._args = [bool2int(a) if a.dtype is NativeBool() else a for a in args]
```

Comment:
The function `bool2int` carries out the same check done in this for loop. Can we not write `self._args = [Int(a) if a.dtype is NativeBool() else a for a in args]`?